hibernate-core/src/main/java/org/hibernate/collection/internal/PersistentBag.java

```diff
@@ -32,8 +32,12 @@
  */
 public class PersistentBag extends AbstractPersistentCollection implements List {
 
+	// TODO: Why is this.bag protected? Can it be changed to private?
 	protected List bag;
 
+	// The Collection provided to a PersistentBag constructor,
```

Comment:
Not a huge fan of storing this - but I guess this gets to the overall solution you are trying.    See my general review comment...